src/ai/ai.cpp

```diff
@@ -1801,13 +1847,15 @@ void update_cb_fabrication(sys::state& state) {
 }
 
 bool will_join_war(sys::state& state, dcon::nation_id n, dcon::war_id w, bool as_attacker) {
-	if(!as_attacker)
-		return true;
 	for(auto par : state.world.war_get_war_participant(w)) {
```

Comment:
Put the removed condition back. It ensures that the AI will answer defensive CTA. Without this guarantee, alliances are meaningless

Reply:
I'm troubled as to why we are allowing alliances to be so meaningful, even players break alliances all the time - bounding AI to be a faithful ally is detrimental to itself, it will help in wars it doesn't care about and really would any player do the same?